packages/api/src/router/photo.ts

```diff
@@ -0,0 +1,185 @@
+          }),
+        );
+      }),
+      */
+
+  addPhotosToReview: protectedProcedure
+    .input(CreatePhotosSchemaExternal)
+    .mutation(async ({ ctx, input }) => {
+      // If the user does not own the review, or the review does not exist,
+      // do not move their file to the prod bucket
+      const review = await ctx.db.query.review.findFirst({
+        where: eq(schema.review.id, input.reviewId),
+      });
+      if (!review) {
+        throw new Error("Review not found");
+      }
+      if (review.userId !== ctx.session.user.id) {
+        throw new Error("Not authorized");
+      }
+
+      const filenames = [] as string[];
+      // Move every file to prod bucket
+      // TODO: verify signedUrls and size are same length
+      for (let i = 0; i < input.signedUrls.length; i++) {
+        const signedUrl = input.signedUrls[i];
+        const size = input.sizes[i];
+        if (!signedUrl || !size) {
+          throw new Error("signedUrl and size query parameters are required");
+        }
+        const res = await fetch(
+          getBaseUrl() +
+            `/api/photo?action=move&signedUrl=${encodeURIComponent(
+              signedUrl,
+            )}&size=${size}`,
+        );
+
+        const { success } = (await res.json()) as { success: boolean };
+        if (success) {
+          // This is safe as long as the "move" action in the above fetch verifies the URL
+          const url = new URL(signedUrl);
+          const filename = url.pathname.split("/").pop();
+          if (filename) {
+            filenames.push(filename);
+          }
+        }
+      }
+      addPhotosToReviewInDB(ctx, input.reviewId, filenames).catch((e) => {
+        throw e;
+      });
+    }),
+
+  // TODO: needs to also delete the files from the production storage bucket
+  // delete should only be called by the user who owns the review
+  delete: protectedProcedure
+    .input(z.string().uuid())
+    .mutation(async ({ ctx, input }) => {
+      const photo = await ctx.db.query.photo.findFirst({
+        where: eq(schema.photo.id, input),
+      });
+      if (!photo) {
+        throw new Error("Photo not found");
+      }
+      const review = await ctx.db.query.review.findFirst({
+        where: eq(schema.review.id, photo.reviewId),
+      });
+      if (!review) {
+        // This would be really bad...
+        throw new Error("Review not found");
+      } else if (review.userId !== ctx.session.user.id) {
+        throw new Error("Not authorized");
+      }
+      await ctx.db.delete(schema.photo).where(eq(schema.photo.id, input));
+    }),
+});
+
+// PRECONDITION: The user must own the review
+async function addPhotosToReviewInDB(
+  ctx: {
+    session: {
+      user: {
+        id: string;
+      };
+      expires: string;
+    };
+    db: Database;
+  },
+  reviewId: number,
+  photoIds: string[],
+) {
+  interface row {
+    reviewId: number;
+    id: string;
+    createdAt: Date;
+  }
+  const rows = [] as row[];
```

Comment:
```suggestion
  const rows: row[] = [];
```
don't use "as", prefer colon (should search your code for other uses)